### PR TITLE
Add the ObjectContent HttpContent Type

### PR DESCRIPTION
**Pull request by yiglas**
Add the ObjectContent and allow the passing of any type to the `ReturnsAsync`. This makes it easier when you are working with JSON object returns from a RESTful service

### CODE REVIEW

File: src/MockNet/Http/Contents/ObjectContent.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            {
+                return false;
+            }
+
+            if (obj.GetType() == type)
+            {
+                return this.value.Equals(obj);
+            }
+
+            {
```
Comment (bpruitt-goddard) [inline src/MockNet/Http/Contents/ObjectContent.cs:49]: Extra brackets here and on line 56?

**Comment by bpruitt-goddard** [comment]
Can the commits be squashed or have one be renamed? They are confusing with being near identically named.

File: src/MockNet/Http/Contents/ObjectContent.cs (hunk 1 above)
Comment (yiglas) [reply src/MockNet/Http/Contents/ObjectContent.cs:49]: Actually, I put them there so I can use the same variable `content` in two different sections.

File: src/MockNet/Http/Contents/ObjectContent.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Text;
+using SystemHttpContent = System.Net.Http.HttpContent;
+using SystemStringContent = System.Net.Http.StringContent;
+
+namespace Theorem.MockNet.Http
+{
+    public class ObjectContent : IHttpContent
```
Comment (inkel) [inline src/MockNet/Http/Contents/ObjectContent.cs:8]: Given the media type is fixed to JSON, shouldn't this be called `JsonObjectContent` in that case?

File: src/MockNet/Http/Contents/ObjectContent.cs (hunk 1 above)
Comment (inkel) [reply src/MockNet/Http/Contents/ObjectContent.cs:49]: If we're targetting C# 8, these two scopes and the final `return` could be replaced with:

```csharp
            var other = obj switch
            {
                ObjectContent c => c.value,
                SystemStringContent c => Utils.Json.ToObject(c.ReadAsStringAsync().GetAwaiter().GetResult(), type),
                _ => throw new NotImplementedException(),
            };

            return this.value.Equals(other);
```

I imagine that it's even possible to add the `null` and `type == obj.Gettype()` comparisons to the statement too.

File: tests/MockNet.Tests/Company.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+using System;
+using System.Collections.Generic;
+
+namespace Theorem.MockNet.Http.Tests
+{
+    public class Company
+    {
+        public IEnumerable<Employee> Employees { get; set; }
+    }
+}
```
Comment (inkel) [inline tests/MockNet.Tests/Company.cs:10]: Newline :wink:

File: tests/MockNet.Tests/Content/ObjectContentTests.cs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        [Fact]
+        public void GenericContentEqualsNullObjectReturnsFalse()
+        {
+            var content = new ObjectContent(typeof(string), "expected");
+
+            Assert.False(content.Equals(null));
+        }
+
+    }
+}
```
Comment (inkel) [inline tests/MockNet.Tests/Content/ObjectContentTests.cs:95]: Newline.

File: tests/MockNet.Tests/Employee.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+            return this == obj;
+        }
+
+        public override int GetHashCode()
+        {
+            return $"{Id}-{Name}-{HireDate}-{WorkHours}".GetHashCode();
+        }
+    }
+}
```
Comment (inkel) [inline tests/MockNet.Tests/Employee.cs:42]: Newline.

File: src/MockNet/Http/Contents/ObjectContent.cs (hunk 1 above)
Comment (inkel) [reply src/MockNet/Http/Contents/ObjectContent.cs:49]: Actually, the following works and passes all tests:

```csharp
           var other = obj switch
            {
                ObjectContent c => c.value,
                SystemStringContent c => Utils.Json.ToObject(c.ReadAsStringAsync().GetAwaiter().GetResult(), type),
                _ => obj,
            };

            return value.Equals(other);
```

Comment (yiglas) [reply src/MockNet/Http/Contents/ObjectContent.cs:49]: I like the switch statement expression here. switching.

Comment (yiglas) [reply src/MockNet/Http/Contents/ObjectContent.cs]: Updated

File: tests/MockNet.Tests/Company.cs (hunk 3 above)
Comment (yiglas) [reply tests/MockNet.Tests/Company.cs]: Added

File: tests/MockNet.Tests/Employee.cs (hunk 5 above)
Comment (yiglas) [reply tests/MockNet.Tests/Employee.cs]: Added

File: tests/MockNet.Tests/Content/ObjectContentTests.cs (hunk 4 above)
Comment (yiglas) [reply tests/MockNet.Tests/Content/ObjectContentTests.cs]: Added

File: src/MockNet/Http/Contents/ObjectContent.cs (hunk 2 above)
Comment (yiglas) [reply src/MockNet/Http/Contents/ObjectContent.cs:8]: Per slack thread leaving the name as ObjectCotent with the default of `application/json` media typ.

**Pull request merged**